### PR TITLE
Flag to avoid unnecesary reindexes

### LINKED ISSUE #3418: Changing property type while reusing title fails
Reproduce:
- blank-state
- add a text property (titled text) to default template and save
- edit default template, delete the text property
- create a numeric property and title it "text"
- elastic mapping error appears

### PR DESCRIPTION
**Pull request by konzz**
fixes #3418

PR checklist:
- [ ] Update READ.me ?
- [ ] Update API documentation ?

QA checklist:
- [ ] Smoke test the functionality described in the issue
- [ ] Test for side effects
- [ ] UI responsiveness
- [ ] Cross browser testing
- [ ] Code review

### CODE REVIEW

**Review by mfacar — CHANGES_REQUESTED** [review]
@konzz, I left a couple of comments, please review them.

Regarding the functional tests, when there is an entity for the template, after accepting to continue from the 'Template conflict' warning, the request never ends, the server however continues working.

File: app/api/entities/entities.js [hunk 1]
```diff
@@ -474,7 +475,7 @@ export default {
   },
 
   /** Rebuild relationship-based metadata objects as {value = id, label: title}. */
-  async updateMetdataFromRelationships(entities, language) {
+  async updateMetdataFromRelationships(entities, language, reindex) {
```
[the remark this answers is not in the document]
Comment (konzz) [reply app/api/entities/entities.js]: yep, added

File: app/api/entities/entities.js [hunk 2]
```diff
@@ -540,10 +544,15 @@ export default {
...
+    if (!template.properties.find(p => p.type === propertyTypes.relationship) && reindex) {
       return search.indexEntities({ template: template._id });
     }
-    return this.bulkUpdateMetadataFromRelationships({ template: template._id, language }, language);
+
+    return this.bulkUpdateMetadataFromRelationships(
+      { template: template._id, language },
+      language,
+      reindex
+    );
```
[the remark this answers is not in the document]
Comment (konzz) [reply app/api/entities/entities.js]: fixed, this was a bug in some scenarios

**Comment by konzz** [comment]
> @konzz, I left a couple of comments, please review them.
> 
> Regarding the functional tests, when there is an entity for the template, after accepting to continue from the 'Template conflict' warning, the request never ends, the server however continues working.

Hi Mercy, I can't replicate that; one of the code issues you found was a possible bug, so maybe that was happening to you? Lets try to have a call and figure it out please.

**Comment by mfacar** [comment]
The issue was fixed with the last commit.  LGTM!

**Pull request merged**